[BUG] maven build error on philter-app

### What is the bug?
Compilation error when running `mvn clean install` on philterd/philter due to phileas package not existing.

`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project philter-app: Compilation failure: Compilation failure:`
`[ERROR] philter/philter-app/src/main/java/ai/philterd/philter/api/controllers/FilterApiController.java:[18,39] package ai.philterd.phileas.model.enums does not exist
` ...etc

### How can one reproduce the bug?
execute `mvn clean install` on philtered/philter 

### What is the expected behavior?
Successful compilation due to matching package/class names within philter-app referencing phileas core.

### What is your host/environment?
MacOS

### Do you have any screenshots?
See stack trace above instead

### Do you have any additional context?
Appears to be due to refactoring in phileas: https://github.com/philterd/phileas/commit/c3af3d7cacbc23736dabdf3b9d18e29bda32a5ce

Thank you!!

--- a/philter-app/src/main/java/ai/philterd/philter/api/controllers/ExplainApiController.java
+++ b/philter-app/src/main/java/ai/philterd/philter/api/controllers/ExplainApiController.java
@@ -15,8 +15,8 @@
  */
 package ai.philterd.philter.api.controllers;
 
-import ai.philterd.phileas.model.enums.MimeType;
-import ai.philterd.phileas.model.objects.FilterResponse;
+import ai.philterd.phileas.model.filtering.FilterResult;
+import ai.philterd.phileas.model.filtering.MimeType;
 import ai.philterd.phileas.policy.Policy;
 import ai.philterd.phileas.services.filters.FilterService;
 import ai.philterd.philter.services.policies.PolicyService;
@@ -49,18 +49,15 @@ public class ExplainApiController extends AbstractController {
 	@RequestMapping(value="/api/explain", method=RequestMethod.POST, produces = MediaType.APPLICATION_JSON_VALUE, consumes = MediaType.TEXT_PLAIN_VALUE)
 	public @ResponseBody ResponseEntity<String> explainTextPlainAsApplicationJson(
 			@RequestParam(value="c", defaultValue="none") String context,
-			@RequestParam(value="d", defaultValue="") String documentId,
 			@RequestParam(value="p", defaultValue="default") String policyName,
 			@RequestBody String body) throws Exception {
 
 			final Policy policy = policyService.get(policyName);
-			final FilterResponse response = filterService.filter(policy, context, documentId, body, MimeType.TEXT_PLAIN);
+			final FilterResult response = filterService.filter(policy, context, body, MimeType.TEXT_PLAIN);
 
 		return ResponseEntity.status(HttpStatus.OK)
-				.header("x-document-id", response.getDocumentId())
 				.contentType(MediaType.APPLICATION_JSON)
 				.body(gson.toJson(response));
-
 
 	}
 

--- a/philter-app/src/main/java/ai/philterd/philter/api/controllers/FilterApiController.java
+++ b/philter-app/src/main/java/ai/philterd/philter/api/controllers/FilterApiController.java
@@ -15,9 +15,9 @@
  */
 package ai.philterd.philter.api.controllers;
 
-import ai.philterd.phileas.model.enums.MimeType;
-import ai.philterd.phileas.model.objects.BinaryDocumentFilterResponse;
-import ai.philterd.phileas.model.objects.FilterResponse;
+import ai.philterd.phileas.model.filtering.BinaryDocumentFilterResult;
+import ai.philterd.phileas.model.filtering.FilterResult;
+import ai.philterd.phileas.model.filtering.MimeType;
 import ai.philterd.phileas.policy.Policy;
 import ai.philterd.phileas.services.filters.FilterService;
 import ai.philterd.philter.services.policies.PolicyService;
@@ -60,10 +60,9 @@ public class FilterApiController extends AbstractController {
 		LOGGER.info("Received uploaded binary PDF file to be returned as ZIP.");
 
 		final Policy policy = policyService.get(policyName);
-		final BinaryDocumentFilterResponse response = filterService.filter(policy, context, documentId, body, MimeType.APPLICATION_PDF, MimeType.IMAGE_JPEG);
+		final BinaryDocumentFilterResult response = filterService.filter(policy, context, body, MimeType.APPLICATION_PDF, MimeType.IMAGE_JPEG);
 
 		return ResponseEntity.status(HttpStatus.OK)
-				.header("x-document-id", response.getDocumentId())
 				.body(response.getDocument());
 
 	}
@@ -78,10 +77,9 @@ public class FilterApiController extends AbstractController {
 		LOGGER.info("Received uploaded binary PDF file to be returned as PDF.");
 
 		final Policy policy = policyService.get(policyName);
-		final BinaryDocumentFilterResponse response = filterService.filter(policy, context, documentId, body, MimeType.APPLICATION_PDF, MimeType.APPLICATION_PDF);
+		final BinaryDocumentFilterResult response = filterService.filter(policy, context, body, MimeType.APPLICATION_PDF, MimeType.APPLICATION_PDF);
 
 		return ResponseEntity.status(HttpStatus.OK)
-				.header("x-document-id", response.getDocumentId())
 				.body(response.getDocument());
 
 	}
@@ -94,10 +92,9 @@ public class FilterApiController extends AbstractController {
             @RequestBody String body) throws Exception {
 
 		final Policy policy = policyService.get(policyName);
-		final FilterResponse response = filterService.filter(policy, context, documentId, body, MimeType.TEXT_PLAIN);
+		final FilterResult response = filterService.filter(policy, context, body, MimeType.TEXT_PLAIN);
 
 		return ResponseEntity.status(HttpStatus.OK)
-				.header("x-document-id", response.getDocumentId())
 				.body(response.getFilteredText());
 
 	}

--- a/philter-app/src/main/java/ai/philterd/philter/api/model/ResponseSpan.java
+++ b/philter-app/src/main/java/ai/philterd/philter/api/model/ResponseSpan.java
@@ -15,8 +15,7 @@
  */
 package ai.philterd.philter.api.model;
 
-import ai.philterd.phileas.model.objects.Span;
-
+import ai.philterd.phileas.model.filtering.Span;
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 
@@ -24,7 +23,7 @@ import java.util.LinkedList;
 import java.util.List;
 
 /**
- * A {@link Span span} returnable to the client.
+ * A span returnable to the client.
  */
 public final class ResponseSpan {
 
@@ -32,18 +31,16 @@ public final class ResponseSpan {
     private int characterEnd;
     private String filterType;
     private String context;
-    private String documentId;
     private double confidence;
     private String replacement;
     private String salt;
 
-    private ResponseSpan(int characterStart, int characterEnd, String filterType, String context, String documentId, double confidence, String replacement, String salt) {
+    private ResponseSpan(int characterStart, int characterEnd, String filterType, String context, double confidence, String replacement, String salt) {
 
         this.characterStart = characterStart;
         this.characterEnd = characterEnd;
         this.filterType = filterType;
         this.context = context;
-        this.documentId = documentId;
         this.confidence = confidence;
         this.replacement = replacement;
         this.salt = salt;
@@ -57,7 +54,6 @@ public final class ResponseSpan {
                 span.getCharacterEnd(),
                 span.getFilterType().toString(),
                 span.getContext(),
-                span.getDocumentId(),
                 span.getConfidence(),
                 span.getReplacement(),
                 span.getSalt()
@@ -86,7 +82,6 @@ public final class ResponseSpan {
                 append(filterType).
                 append(confidence).
                 append(context).
-                append(documentId).
                 append(replacement).
                 append(salt).
                 toHashCode();
@@ -107,7 +102,6 @@ public final class ResponseSpan {
                 + " characterEnd: " + characterEnd
                 + " filterType: " + filterType
                 + " context: " + context
-                + " documentId: " + documentId
                 + " confidence: " + confidence
                 + " replacement: " + replacement
                 + " salt: " + salt;
@@ -144,14 +138,6 @@ public final class ResponseSpan {
 
     public void setContext(String context) {
         this.context = context;
-    }
-
-    public String getDocumentId() {
-        return documentId;
-    }
-
-    public void setDocumentId(String documentId) {
-        this.documentId = documentId;
     }
 
     public double getConfidence() {

--- a/philter-app/src/main/java/ai/philterd/philter/services/PhilterService.java
+++ b/philter-app/src/main/java/ai/philterd/philter/services/PhilterService.java
@@ -16,16 +16,21 @@
 package ai.philterd.philter.services;
 
 import ai.philterd.phileas.PhileasConfiguration;
-import ai.philterd.phileas.model.objects.BinaryDocumentFilterResponse;
-import ai.philterd.phileas.model.objects.FilterResponse;
+import ai.philterd.phileas.model.filtering.ApplyResult;
+import ai.philterd.phileas.model.filtering.BinaryDocumentFilterResult;
+import ai.philterd.phileas.model.filtering.FilterResult;
+import ai.philterd.phileas.model.filtering.MimeType;
+import ai.philterd.phileas.model.filtering.Span;
 import ai.philterd.phileas.policy.Policy;
 import ai.philterd.phileas.services.context.ContextService;
 import ai.philterd.phileas.services.disambiguation.vector.VectorService;
 import ai.philterd.phileas.services.filters.FilterService;
-import ai.philterd.phileas.model.enums.MimeType;
 import ai.philterd.phileas.services.PhileasFilterService;
+import org.apache.commons.lang3.NotImplementedException;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+
+import java.util.List;
 
 @Component
 public class PhilterService implements FilterService {
@@ -38,13 +43,19 @@ public class PhilterService implements FilterService {
     }
 
     @Override
-    public FilterResponse filter(Policy policy, String context, String documentId, String input, MimeType mimeType) throws Exception {
-        return phileasFilterService.filter(policy, context, documentId, input, mimeType);
+    public FilterResult filter(Policy policy, String context, String input, MimeType mimeType) throws Exception {
+        return phileasFilterService.filter(policy, context, input, mimeType);
     }
 
     @Override
-    public BinaryDocumentFilterResponse filter(Policy policy, String context, String documentId, byte[] input, MimeType mimeType, MimeType outputMimeType) throws Exception {
-        return phileasFilterService.filter(policy, context, documentId, input, mimeType, outputMimeType);
+    public BinaryDocumentFilterResult filter(Policy policy, String context, byte[] input, MimeType mimeType, MimeType outputMimeType) throws Exception {
+        return phileasFilterService.filter(policy, context, input, mimeType, outputMimeType);
     }
+
+    @Override
+    public ApplyResult apply(final List<Span> spans, final String input) {
+        throw new NotImplementedException();
+    }
+
 
 }

--- a/philter-app/src/test/java/api/ApiControllerTests.java
+++ b/philter-app/src/test/java/api/ApiControllerTests.java
@@ -1,9 +1,9 @@
 package api;
 
 import ai.philterd.phileas.PhileasConfiguration;
-import ai.philterd.phileas.model.enums.MimeType;
-import ai.philterd.phileas.model.objects.Explanation;
-import ai.philterd.phileas.model.objects.FilterResponse;
+import ai.philterd.phileas.model.filtering.Explanation;
+import ai.philterd.phileas.model.filtering.FilterResult;
+import ai.philterd.phileas.model.filtering.MimeType;
 import ai.philterd.phileas.policy.Policy;
 import ai.philterd.phileas.services.filters.FilterService;
 import ai.philterd.philter.api.controllers.FilterApiController;
@@ -94,7 +94,7 @@ public class ApiControllerTests {
     public void filterTest() throws Exception {
 
         final Policy policy = new Policy();
-        when(filterService.filter(policy, "George Washington was president.", "none", "documentid", MimeType.TEXT_PLAIN)).thenReturn(new FilterResponse("*** was president.", "none", "none", 0, new Explanation(Collections.emptyList(), Collections.emptyList()), Collections.emptyMap(), Collections.emptyList(), 0));
+        when(filterService.filter(policy, "George Washington was president.", "none", MimeType.TEXT_PLAIN)).thenReturn(new FilterResult("*** was president.", "none", 0, new Explanation(Collections.emptyList(), Collections.emptyList()), Collections.emptyList(), 0));
 
         mockMvc.perform(post("/api/filter").content("George Washington was president.").param("c", "none").param("d", "none").contentType("text/plain")).andExpect(status().isOk());
 

--- a/philter-app/src/test/java/api/model/ResponseSpanTest.java
+++ b/philter-app/src/test/java/api/model/ResponseSpanTest.java
@@ -1,7 +1,7 @@
 package api.model;
 
-import ai.philterd.phileas.model.enums.FilterType;
-import ai.philterd.phileas.model.objects.Span;
+import ai.philterd.phileas.model.filtering.FilterType;
+import ai.philterd.phileas.model.filtering.Span;
 import ai.philterd.philter.api.model.ResponseSpan;
 import com.google.gson.Gson;
 import nl.jqno.equalsverifier.EqualsVerifier;
@@ -28,9 +28,9 @@ public class ResponseSpanTest {
 
         final String[] window = new String[]{};
 
-        final Span span1 = Span.make(1, 6, FilterType.AGE, "context", "document", 1.0, "test", "***", "", false, true, window, 0);
-        final Span span2 = Span.make(8, 12, FilterType.AGE, "context", "document", 1.0, "test", "***", "", false, true, window, 0);
-        final Span span3 = Span.make(14, 20, FilterType.AGE, "context", "document", 1.0, "test", "***", "", false, true, window, 0);
+        final Span span1 = Span.make(1, 6, FilterType.AGE, "context", 1.0, "test", "***", "", false, true, window, 0);
+        final Span span2 = Span.make(8, 12, FilterType.AGE, "context", 1.0, "test", "***", "", false, true, window, 0);
+        final Span span3 = Span.make(14, 20, FilterType.AGE, "context",  1.0, "test", "***", "", false, true, window, 0);
 
         final List<Span> spans = Arrays.asList(span1, span2, span3);
 
@@ -42,7 +42,6 @@ public class ResponseSpanTest {
             Assert.assertEquals(spans.get(i).getCharacterEnd(), responseSpans.get(i).getCharacterEnd());
             Assert.assertEquals(spans.get(i).getConfidence(), responseSpans.get(i).getConfidence(), 0);
             Assert.assertEquals(spans.get(i).getContext(), responseSpans.get(i).getContext());
-            Assert.assertEquals(spans.get(i).getDocumentId(), responseSpans.get(i).getDocumentId());
             Assert.assertEquals(spans.get(i).getReplacement(), responseSpans.get(i).getReplacement());
             Assert.assertEquals(spans.get(i).getFilterType().toString(), responseSpans.get(i).getFilterType());
 

